src/dfx/src/lib/operations/canister/create_canister.rs

```diff
@@ -76,7 +80,23 @@ pub async fn create_canister(
     let agent = env.get_agent();
     let cid = match call_sender {
         CallSender::SelectedId => {
-            create_with_management_canister(env, agent, with_cycles, specified_id, settings).await
+            if !CYCLES_LEDGER_ENABLED
+                || (!env.get_network_descriptor().is_ic
+                    && std::env::var("DFX_DISABLE_AUTO_WALLET").is_err())
```

Comment:
Suggestion for readability.  What do you think?
```suggestion
            let cycles_ledger_disabled = !CYCLES_LEDGER_ENABLED;
            let ic_network = env.get_network_descriptor().is_ic;
            let enable_auto_wallet = std::env::var("DFX_DISABLE_AUTO_WALLET").is_err();

            if cycles_ledger_disabled || !ic_network || enable_auto_wallet
```

Reply:
This is now 
```
let auto_wallet_disabled = std::env::var("DFX_DISABLE_AUTO_WALLET").is_ok();
let ic_network = env.get_network_descriptor().is_ic;
if CYCLES_LEDGER_ENABLED && (ic_network || auto_wallet_disabled)
```
WDYT?

Reply:
Are they equivalent?

Regardless, see if you can reduce it to three terms and a single logic type.

"A && B && C" is going to be easier to read and understand than "A && (B || D)"